gammapy/data/data_store.py

```diff
@@ -284,6 +284,43 @@ def load_many(self, obs_ids, hdu_type=None, hdu_class=None):
 
         return things
 
+    def check_observations(self):
+        """Perform some sanity checks for all observations.
+
+        Returns
+        -------
+        OrderedDict : dictionary containing failure messages for all runs that fail a check.
+        """
+
+        results = OrderedDict()
+        nfail = 0
+
+        # Loop over all obs_ids in obs_table
+        for obs_id in self.obs_table['OBS_ID']:
+            messages = []
```

Comment:
Everything in the for loop should be moved to a separate method. Would it fit better on the `obs` class?